### ISSUE TITLE
V5 Regression: Incorrect value when scanning NUMERIC into `uint64`

### ISSUE THREAD

**Issue by uberbrodt** [jackc/pgx#1414]
**Describe the bug**
Scanning values into a uint64 from a NUMERIC leads to incorrect values. This was working in `v4`

**To Reproduce**

If possible, please provide runnable example such as:

```go
package main

import (
	"context"
	"log"

	"github.com/jackc/pgx/v5"
)

func main() {
	// CREATE TABLE test_uint64 (
	//     partition_key NUMERIC(22,0) NOT NULL,
	//     sort          BIGSERIAL
	// );

	// Test data

	// [Ï]dev=> select * from test_uint64;
	//  partition_key | sort
	// ---------------+------
	//              5 |    1
	//            100 |    2
	//         100000 |    3
	// (3 rows)

	ctx := context.Background()
	conn, err := pgx.Connect(ctx, "")
	if err != nil {
		log.Fatal(err)
	}

	sql := `
	SELECT partition_key
		FROM test_uint64 where partition_key > $1 OR partition_key < $2
	ORDER BY sort ASC
	`

	defer conn.Close(context.Background())

	if rows, err := conn.Query(ctx, sql, 0, 100); err != nil {
		log.Fatal(err)
	} else {
		defer rows.Close()
		for rows.Next() {
			var partitionKey uint64

			err := rows.Scan(&partitionKey)
			if err != nil {
				log.Fatal(err)
			}
			log.Printf("Fetch partitionKey: %d", partitionKey)

		}
	}
}

}
```

**Expected behavior**
I expected the output to be this (this is the program above run with partitionKey set to `int64`):
```
}2022/12/05 16:38:28 Fetch partitionKey: 5
}2022/12/05 16:38:28 Fetch partitionKey: 100
}2022/12/05 16:38:28 Fetch partitionKey: 100000
```

**Actual behavior**
I received this instead:
```
}2022/12/05 16:37:45 Fetch partitionKey: 5
}2022/12/05 16:37:45 Fetch partitionKey: 1
}2022/12/05 16:37:45 Fetch partitionKey: 1
```

**Version**
 - Go: `go version go1.19.2 darwin/arm64`
 - PostgreSQL: `$ PostgreSQL 13.9 (Debian 13.9-1.pgdg110+1) on aarch64-unknown-linux-gnu, compiled by gcc (Debian 10.2.1-6) 10.2.1 20210110, 64-bit`
 - pgx: `github.com/jackc/pgx/v5 v5.1.1`

**Additional context**

**Comment by jackc** [comment]
Fix is in newly released `v5.2.0`.

**Comment by uberbrodt** [comment]
Thanks for the timely fix and all the work you do!
